core/src/main/java/hudson/model/Computer.java

```diff
@@ -1369,15 +1369,22 @@ public String call() throws IOException {
 //
 //
     public void doRssAll( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException {
-        rss(req, rsp, " all builds", getBuilds());
+        RSS.rss(req, rsp, getDisplayName() + " all builds", getUrl(), getBuilds());
     }
 
     public void doRssFailed(StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException {
-        rss(req, rsp, " failed builds", getBuilds().failureOnly());
+        RSS.rss(req, rsp, getDisplayName() + " failed builds", getUrl(), getBuilds().failureOnly());
     }
-    private void rss(StaplerRequest req, StaplerResponse rsp, String suffix, RunList runs) throws IOException, ServletException {
-        RSS.forwardToRss(getDisplayName() + suffix, getUrl(),
-                runs.newBuilds(), Run.FEED_ADAPTER, req, rsp);
+
+    public void doRssLatest( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException {
+        final List<Run> lastBuilds = new ArrayList<>();
+        for (AbstractProject<?, ?> p : Jenkins.get().allItems(AbstractProject.class)) {
```

Comment:
Should the limitation to `AbstractBuild` be documented somewhere?

Reply:
I've just wanted to keep some coherence and I just saw  https://github.com/jenkinsci/jenkins/blob/4d059825ad8595f23f5e9396091d53e8780d4065/core/src/main/java/hudson/model/User.java#L882-L896

TBH, I don't know if any kind of limitation should apply. I will add a javadoc explaining what is displayed and if you think something else / other documentation is needed just let me know and I'm happy to add it.

Reply:
I think now I've understood you well ...

Reply:
Maybe [5a2c62e](https://github.com/jenkinsci/jenkins/pull/4411/commits/5a2c62e5c72320ba8ba9fd9aff4f514e0e385249)?

Reply:
I think Oleg meant that this code only takes into account `AbstractProject`s, it doesn't take into account, for example, `FreeStyleProject`s.

Reply:
It does: `FreeStyleProject` extends from `Project` ([see here](https://github.com/jenkinsci/jenkins/blob/ad1ca7101b9b180dc677eef914b1cbd8208d00c8/core/src/main/java/hudson/model/FreeStyleProject.java#L40)) which extends from `AbstractProject` ([see here](https://github.com/jenkinsci/jenkins/blob/ad1ca7101b9b180dc677eef914b1cbd8208d00c8/core/src/main/java/hudson/model/Project.java#L64-L65)).
My local testing was using a pipeline and a FreeStyle Project and both sort of projects appeared in the feed. As I said above, I'm mimicking the behaviour in `User` class. I'm totally happy with adding whatever is missing or with documenting whatever is needed, but I need more detail.

Reply:
Would performance be a concern here? Given we are asking for all the items and looping through them?

Reply:
TBH, I don't think this feature is widely used since it's been broken for so long, so personally I wound't care (personal opinion).
As I said above, I'm mimicking the current behaviour for the already existing `User` class. If there is another way to gather that information, I'm unknown. And also I'm open to suggestions if this is not the information that should be retrieved.

Reply:
Sounds reasonable. A bad performant feature (worst case scenario) is better than a broken one.

Reply:
> I think Oleg meant that this code only takes into account AbstractProjects, it doesn't take into account, for example, FreeStyleProjects.

It does not take Pipeline into account, for example. I know that we are still waiting for proper Node information API to be exposed for Pipeline, but 🤷‍♂